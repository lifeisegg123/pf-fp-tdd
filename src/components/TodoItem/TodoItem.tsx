import { MouseEventHandler } from "react";
import { Button } from "../Button";
import { Typo } from "../Typo";

interface TodoItemProps {
  desc: string;
  checked: boolean;
  onRemove: MouseEventHandler<HTMLButtonElement>;
  onCheck: MouseEventHandler<HTMLLIElement>;
}

export function TodoItem({ desc, checked, onRemove, onCheck }: TodoItemProps) {
  const handleRemove: TodoItemProps["onRemove"] = (e) => {
    e.stopPropagation();
    onRemove(e);
  };
  return (
    <li className="flex-row-center gap-sm" onClick={onCheck}>
      <span>{checked ? "done" : "todo"}</span>
      <Typo as="h2">{desc}</Typo>
      <Button onClick={handleRemove} aria-label="삭제">
        X
      </Button>
    </li>
  );
}
