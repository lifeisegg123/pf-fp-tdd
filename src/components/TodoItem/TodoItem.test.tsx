import { TodoItem } from "./TodoItem";
import { fireEvent, render } from "@testing-library/react";

describe("Component/TodoItem", () => {
  it("should match snapshot", () => {
    const mockRemove = jest.fn();
    const mockCheck = jest.fn();
    const { container } = render(
      <TodoItem desc="test" checked onRemove={mockRemove} onCheck={mockCheck} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render description", () => {
    const mockRemove = jest.fn();
    const mockCheck = jest.fn();
    const { getByText } = render(
      <TodoItem desc="test" checked onRemove={mockRemove} onCheck={mockCheck} />
    );

    expect(getByText("test")).toBeInTheDocument();
  });

  it("should handle click remove button", () => {
    const mockRemove = jest.fn();
    const mockCheck = jest.fn();
    const { getByLabelText } = render(
      <TodoItem desc="test" checked onRemove={mockRemove} onCheck={mockCheck} />
    );

    fireEvent.click(getByLabelText("삭제"));

    expect(mockRemove).toBeCalled();
    expect(mockCheck).not.toBeCalled();
  });
});
