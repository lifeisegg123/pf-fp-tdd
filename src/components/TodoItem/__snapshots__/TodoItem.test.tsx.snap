// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/TodoItem should match snapshot 1`] = `
<div>
  <li
    class="flex-row-center gap-sm"
  >
    <span>
      done
    </span>
    <h2>
      test
    </h2>
    <button
      aria-label="삭제"
    >
      X
    </button>
  </li>
</div>
`;
